spring-cloud-function-samples/function-sample-spring-integration/src/main/java/example/FunctionSampleSpringIntegrationApplication.java

```diff
@@ -26,21 +26,21 @@
 import org.springframework.integration.handler.LoggingHandler;
 import org.springframework.messaging.Message;
 
-//@SpringBootApplication
+@SpringBootApplication
 public class FunctionSampleSpringIntegrationApplication {
 
 	public static void main(String[] args) {
 		SpringApplication.run(FunctionSampleSpringIntegrationApplication.class, args);
 	}
 
-	//@Bean
+	@Bean
 	public IntegrationFlow uppercaseFlow() {
 		return IntegrationFlows.from(MessageFunction.class, "uppercase")
 				.<String, String>transform(String::toUpperCase)
 				.logAndReply(LoggingHandler.Level.WARN);
 	}
 
-	public interface MessageFunction extends Function<Message<String>, Message<String>> {
+	public interface MessageFunction<T> extends Function<Message<String>, Message<String>> {
```

Comment:
Why do we need this `<T>` here?

Reply:
No reason, just wanted to create a deep hierarchy and validate that the new `discover` method still finds the function type.

Reply:
OK, but it must not be as a part of a sample. It is going to confuse end-users.

Reply:
ohh, i forgot to delete it. Good point